src/CMakeLists.txt

```diff
@@ -45,6 +45,10 @@ target_compile_definitions( T8 PUBLIC T8_VERSION_POINT=${T8CODE_VERSION_POINT} )
 target_include_directories( T8 PUBLIC ${CMAKE_CURRENT_LIST_DIR} )
 target_link_libraries( T8 PUBLIC P4EST::P4EST SC::SC )
 
+if ( CMAKE_BUILD_TYPE STREQUAL "Debug" )
```

Comment:
Does this only set the macro for building `libt8.so`? Some tests and some examples also have this macro. Can the debug flag set globally for all targets?